src/Crdt.Sample/Changes/NewWordChange.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Crdt.Changes;
+using Crdt.Core;
+using Crdt.Entities;
+using Crdt.Sample.Models;
+
+namespace Crdt.Sample.Changes;
+
+public class NewWordChange(Guid entityId, string text, string? note = null) : Change<Word>(entityId), ISelfNamedType<NewWordChange>
+{
+    public string Text { get; } = text;
+    public string? Note { get; } = note;
+
+    public override IObjectBase NewEntity(Commit commit)
+    {
+        return new Word
+        {
+            Text = Text,
+            Note = Note,
+            Id = EntityId
+        };
+    }
+
+    public override ValueTask ApplyChange(Word entity, ChangeContext context)
+    {
+        entity.Text = Text;
+        entity.Note = Note;
+        return ValueTask.CompletedTask;
+    }
```

Comment:
Having 2 interfaces is a good idea, I didn't initially have the NewEntity method and so there was only Apply, but I added it later to support required/readonly fields.

Reply:
so I was looking at the code to go and implement this and there's a difficult problem. Let's say there's 3 interfaces. One for all Changes, which we have today which requires a change to be against a specific entity, right now this also has methods for both Apply and NewEntity. If we remove those methods from the base interface and add 2 more, one for New and one for Apply, how do we go about requiring the use of one or the other? They could just implement none which wouldn't make any sense so we should try to make that impossible. That said requiring a `NewEntity` method and telling the dev to just throw a not supported exception also isn't great.

Reply:
Looking at current usage I think it would make sense to add an `EditChange` abstract class that extends `Change` and is still abstract with an implementation of NewEntity that throws NotImplemented. Then developers could extend the EditChange in the appropriate cases.

Reply:
What if we make the base interface internal.

Reply:
Had a chat with Jason about this. There's now an EditChange and CreateChange abstract class, take a look at the sample project to see how I used it. The SetWordChange was used as both, so I've left it inheriting from Change and it still implements both cases as that makes tests much simpler to have a single change for both cases.